aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapterQueryTest.java

```diff
@@ -237,10 +252,57 @@ public void queryWithMaliciousPredicates() throws DataStoreException {
         adapter.save(jane);
 
         QueryPredicate predicate = BlogOwner.NAME.eq("Jane; DROP TABLE Person; --");
-        final List<BlogOwner> resultOfMaliciousQuery = adapter.query(BlogOwner.class, predicate);
+        final List<BlogOwner> resultOfMaliciousQuery = adapter.query(BlogOwner.class, where(predicate));
         assertTrue(resultOfMaliciousQuery.isEmpty());
 
         final List<BlogOwner> resultAfterMaliciousQuery = adapter.query(BlogOwner.class);
         assertTrue(resultAfterMaliciousQuery.contains(jane));
     }
+
+    @Test
+    public void queryWithPaginationWithCustomValues() throws DataStoreException {
+        final int pageSize = 10;
+        createBlogOwnerRecords(pageSize * 2);
+
+        List<BlogOwner> result = adapter.query(
+            BlogOwner.class,
+            all().paginated(page(0).withLimit(pageSize))
```

Comment:
I like this style, but I'd change the method names for the pagination. To me, this would be more clear:
```
Where.allMatch().paginated(startAtPage(0).withPageSize(pageSize))
```

Reply:
> Nice work @drochetti! Overall looks great. Some ideas mostly about naming, inline.

Thanks for the early review @jamesonwilliams. I'll update the PR with your suggestions.

Reply:
I like your suggestion better than the current implementation. My only concern is that JS went GA with the `page`/`limit` [pagination API](https://docs.amplify.aws/lib/datastore/data-access/q/platform/js#pagination). I'm not strongly opposed to change it into something more expressive, but I'd say we should align iOS and Android in that case.